docs/legacy-project-tips.md

```diff
@@ -63,6 +63,44 @@ linting process more manageable, we're going to provide a
 some tips of tricks to break things down into bite size
 chunks.
 
+Pre-commit hook to lint only changed files
+------------------------------------------
+
+Using npm-scripts, some bash-fu and the [husky][husky] 
+module, you can have lint-trap only lint changed *.js files
+as a pre-commit hook. To do this, first install husky as a
+dev dependency:
+
+    $ npm install --save-dev husky
+
+Then add the following to your `package.json` file:
+
+    "scripts": {
+        ...
+        "lint-changed": "lint-trap `git diff HEAD --name-only --diff-filter=ACMRTU | awk '/\.js$/' | awk  -v path=$(git rev-parse --show-toplevel)/ '{print path$1}'`",
+        "precommit": "npm run lint-changed",
+        ...
+    },
+
+Husky will install git-hooks, including a pre-commit hook
```

Comment:
@malandrew I contributed to `pre-commit` to ensure its silent by default unless it fails. I would have to verify that husky doesnt suffer that

 - https://github.com/observing/pre-commit/commit/7a801e6f4ffa0bb0a92db680c6023b72f59e36fb
 - https://github.com/observing/pre-commit/commit/80a6733752983862d5da3c08093dda879c35a1e0
 - https://github.com/observing/pre-commit/commit/4591bf32cd526b08547eab6a71880d123405ddfc
 - https://github.com/observing/pre-commit/commit/a7cbbca67c5f8787183e5df180b71ca1be65168a
 - https://github.com/observing/pre-commit/commit/95cc4f4ba94608d40c99b57bfd91a54b1131f8fb